src/schema/schema_create.c

```diff
@@ -1476,6 +1476,12 @@ __wt_schema_create(WT_SESSION_IMPL *session, const char *uri, const char *config
     WT_DECL_RET;
     WT_SESSION_IMPL *int_session;
 
+    /*
+     * This function seems to not always expect the schema lock to be acquired. For example, the
+     * lock is not acquired if we get here through __wt_tiered_open, but we still have to verify
+     * that.
+     */
```

Comment:
There is no verification here?

Reply:
I wasn't able to confirm that we always call this with the schema lock. I traced calls to this function about 4-5 levels deep, and one of those paths doesn't include acquiring the lock. Maybe it's acquired further up, but it's really hard to verify it as that function 4 or 5 levels up gets called from a large number of places, all without acquiring the lock. That being said, it's possible that this is a bug, but so far I was just documenting the current locking behavior.

Reply:
That being said, I went ahead and added an assert here, even though I wasn't able to verify this through code reading. It passed both the csuite and the Python suite, so I hope it is indeed correct 👍 In the worst case, we'll get a test failure at some point, at which point we'll figure out what is the right thing to do. Needing schema lock at this point sounds reasonable.